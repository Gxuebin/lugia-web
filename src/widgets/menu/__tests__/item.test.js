//@flow

import * as React from 'react';
import chai from 'chai';
import MenuItem from '../item';
import 'jest-styled-components';

import renderer from 'react-test-renderer';

const { expect: exp, } = chai;

describe('Item', () => {

  it('children: null', () => {
    const target = renderer.create(<MenuItem></MenuItem>).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Single MenuItem', () => {
    const defaultItem = renderer.create(<MenuItem>hello</MenuItem>).toJSON();
    const target = renderer.create(<MenuItem checked={false}>hello</MenuItem>).toJSON();
    expect(target).toMatchSnapshot();
    expect(target).toMatchObject(defaultItem);
  });

  it('Single MenuItem checked', () => {
    const target = renderer.create(<MenuItem checked><a>hello</a></MenuItem>);
    expect(target).toMatchSnapshot();
  });
  it('Mutliple MenuItem', () => {
    const defaultItem = renderer.create(<MenuItem mutliple><a>hello</a></MenuItem>).toJSON();
    const target = renderer.create(<MenuItem mutliple checked={false}><a>hello</a></MenuItem>).toJSON();
    expect(target).toMatchSnapshot();
    expect(target).toMatchObject(defaultItem);
  });

  it('Mutliple MenuItem checked', () => {
    const target = renderer.create(<MenuItem mutliple checked><a>hello</a></MenuItem>);
    expect(target).toMatchSnapshot();
  });


});
