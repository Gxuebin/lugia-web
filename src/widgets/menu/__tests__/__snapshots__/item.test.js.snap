// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Mutliple MenuItem 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 35px;
  padding: 7px 8px;
  font-weight: 400;
  color: rgba(0,0,0,0.65);
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c0:hover {
  background-color: #ecf6fd;
}

.c0::after {
  font-family: "sviconfont" !important;
  text-rendering: optimizeLegibility;
  content: "\\e73e";
  color: transparent;
  display: inline-block;
  font-size: 1.2rem;
  -webkit-transform: scale(.83333333) rotate(0deg);
  -ms-transform: scale(.83333333) rotate(0deg);
  transform: scale(.83333333) rotate(0deg);
  zoom: 1;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 8px;
  font-weight: 700;
  text-shadow: 0 0.1px 0,0.1px 0 0,0 -0.1px 0,-0.1px 0;
}

.c0:hover:after {
  color: #d0c8c8;
}

<li
  checked={false}
  className="c0"
  title=""
>
  <a>
    hello
  </a>
</li>
`;

exports[`Item Mutliple MenuItem checked 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 35px;
  padding: 7px 8px;
  font-weight: 400;
  color: rgba(0,0,0,0.65);
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  background-color: #f7f7f7;
  font-weight: 600;
  color: rgba(0,0,0,.65);
}

.c0:hover {
  background-color: #ecf6fd;
}

.c0::after {
  font-family: "sviconfont" !important;
  text-rendering: optimizeLegibility;
  content: "\\e73e";
  color: transparent;
  display: inline-block;
  font-size: 1.2rem;
  -webkit-transform: scale(.83333333) rotate(0deg);
  -ms-transform: scale(.83333333) rotate(0deg);
  transform: scale(.83333333) rotate(0deg);
  zoom: 1;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 8px;
  font-weight: 700;
  text-shadow: 0 0.1px 0,0.1px 0 0,0 -0.1px 0,-0.1px 0;
}

.c0:after {
  color: #108ee9;
}

.c0:hover:after {
  color: #108ee9;
}

<li
  checked={true}
  className="c0"
  title=""
>
  <a>
    hello
  </a>
</li>
`;

exports[`Item Single MenuItem 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 35px;
  padding: 7px 8px;
  font-weight: 400;
  color: rgba(0,0,0,0.65);
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c0:hover {
  background-color: #ecf6fd;
}

<li
  checked={false}
  className="c0"
  title="hello"
>
  hello
</li>
`;

exports[`Item Single MenuItem checked 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 35px;
  padding: 7px 8px;
  font-weight: 400;
  color: rgba(0,0,0,0.65);
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  background-color: #f7f7f7;
  font-weight: 600;
  color: rgba(0,0,0,.65);
}

.c0:hover {
  background-color: #ecf6fd;
}

<li
  checked={true}
  className="c0"
  title=""
>
  <a>
    hello
  </a>
</li>
`;

exports[`Item children: null 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 35px;
  padding: 7px 8px;
  font-weight: 400;
  color: rgba(0,0,0,0.65);
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c0:hover {
  background-color: #ecf6fd;
}

<li
  checked={false}
  className="c0"
  title=""
/>
`;
